src/Plugin/audio_embed_field/Provider/StanfordSoundCloud.php

```diff
@@ -57,6 +56,21 @@ public static function getIdFromInput($input) {
     return (int) end($tracks_parts) ? end($tracks_parts) : NULL;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function renderEmbedCode($width, $height, $autoplay) {
+    $render = parent::renderEmbedCode($width, $height, $autoplay);
+    $video_data = static::getVideoData($this->input);
+
+    // Fix the url if the input is actually a playlist instead of a track.
+    if (strpos($video_data['html'], 'playlist') !== FALSE) {
+      $render['#url'] = str_replace('/tracks/', '/playlists/', $render['#url']);
```

Comment:
That seems like a nice helper, but does it give the user the desired outcome just by swapping out the part in the url? Does the right track play? Could that be confusing?

Reply:
neat.